backend/postgres/db.go

```diff
@@ -125,6 +126,20 @@ const (
 	rollbackClass   = pq.ErrorClass("40")
 )
 
+type queryCanceledError struct {
+	cause *pq.Error
+}
+
+func (qce *queryCanceledError) Error() string {
+	return qce.cause.Error()
+}
+
+func (qce *queryCanceledError) Is(target error) bool {
+	return target == context.Canceled
```

Comment:
Should we fall back to a the wrapped error if the condition failed ?
```suggestion
	if target == context.Canceled {
	  return true
	}
	
	return errors.Is(qce.cause, target)
```
?

Reply:
it is done directly by the std lib in the implementation of `errors.Is`

Reply:
Won't we need the Unwrap method to be implemented on this type then ?